### PR TITLE
Pivot table MVP

### PR DESCRIPTION
**Pull request by mknowlton89**
### Features and Changes

This PR adds a new `Pivot Table` visualization option to the Sql Explorer & Product Analytics Dashboard builder. It currently supports only a single level for columns, rows, and dimensions, though it was built in a manner to where we can extend each to n-dimensions later.

This is an MVP of the feature. Future fast follows will be:
- Additional levels of depth for columns, rows, & dimensions - need to update the UI to support this.
- Configurable colors / enable/disable opacity shading
- Collapsible roll-up rows (maybe - this hasn't been confirmed)

### Testing

- [x] Build out a pivot table with some sample data and ensure the aggregation and transformation is correct
- [x] Ensure that if a column is a `date` then we format it according to the granularity

### CODE REVIEW

File: packages/front-end/components/DataViz/SqlExplorerDataVisualization.tsx [hunk 1]
```diff
@@ -687,6 +842,241 @@ export function DataVisualizationDisplay({
...
+                          fontWeight: row.isBold ? "bold" : "normal",
+                          background:
+                            typeof value === "number" && maxCellValue > 0
+                              ? `color-mix(in srgb, #5071de ${Math.round(
+                                  Math.max(
+                                    0,
+                                    Math.min(1, value / maxCellValue),
+                                  ) * 85,
+                                )}%, transparent)`
+                              : undefined,
```
Comment (mknowlton89) [inline packages/front-end/components/DataViz/SqlExplorerDataVisualization.tsx]: I need to test this with negative values.

File: packages/front-end/services/dataVizConfigUtilities.ts [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+  }
+
+  // For other chart types, store as single object (first element)
+  // If empty array, set to undefined
+  return {
+    ...config,
+    xAxis: (valueArray.length > 0 ? valueArray[0] : undefined) as Partial<
+      BarChart | LineChart | AreaChart | ScatterChart
+    >["xAxis"],
+  } as Partial<DataVizConfig>;
```
Comment (Kevin-Chant) [inline packages/front-end/services/dataVizConfigUtilities.ts]: These two typecasts are unnecessary; I'd remove them both for simplicity and for extra type safety in the future

File: packages/front-end/services/dataVizConfigUtilities.ts [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+}
+
+export function updateXAxisConfig(
+  config: Partial<DataVizConfig>,
+  updates: Partial<xAxisConfiguration>,
+): Partial<DataVizConfig> {
+  const configs = getXAxisConfig(config);
+  const current = configs[0];
```
Comment (Kevin-Chant) [inline packages/front-end/services/dataVizConfigUtilities.ts:55]: This is fine for now because we're only doing a single x axis for pivot tables in the mvp right? I assume we'll pass an index in instead of 0 when we expand it?

File: packages/front-end/components/DataViz/PivotTable.tsx
Comment (Kevin-Chant) [inline packages/front-end/components/DataViz/PivotTable.tsx:1]: Like we said in Slack I think we should move this file wholesale to `packages/front-end/components/SqlExplorer/PivotTable.tsx`. I don't see anything that needs to be in `Dashboards/`

File: packages/front-end/components/DataViz/PivotTable.tsx [hunk 4]
```diff
@@ -0,0 +1,603 @@
...
+          return new Intl.DateTimeFormat(undefined, {
+            year: "numeric",
+          }).format(d);
+        case "none":
+        default:
+          return d.toLocaleDateString();
+      }
+    }
+    return String(xVal);
+  };
```
Comment (Kevin-Chant) [inline packages/front-end/components/DataViz/PivotTable.tsx:131]: Nit: This feels like it should be a helper on its own rather than defined inside another function. You could curry the values of `isDateType` and `xConfig` to preserve the current usage

File: packages/front-end/components/DataViz/SqlExplorerDataVisualization.tsx [hunk 5]
```diff
@@ -282,67 +285,120 @@ export function DataVisualizationDisplay({
     });
   }, [dataVizConfig.filters, rows]);
 
-  // TODO: Support multiple y-axis and dimension fields
-  const xConfig = requiresXAxis(dataVizConfig)
-    ? dataVizConfig.xAxis
-    : undefined;
+  // TODO: Support multiple y-axis fields
+  const xAxisConfigs = getXAxisConfig(dataVizConfig);
+  const xConfig = xAxisConfigs[0];
```
Comment (Kevin-Chant) [inline packages/front-end/components/DataViz/SqlExplorerDataVisualization.tsx:290]: Double checking here as well - is the impact of this 0 index that we cant add multiple X axes yet (and will be changed in a future PR)?

File: packages/front-end/components/DataViz/DataVizConfigPanel.tsx [hunk 6]
```diff
@@ -149,18 +172,19 @@ export default function DataVizConfigPanel({
                 onDataVizConfigChange({
                   ...dataVizConfig,
                   chartType: v as DataVizConfig["chartType"],
-                });
+                } as Partial<DataVizConfig>);
```
Comment (Kevin-Chant) [inline packages/front-end/components/DataViz/DataVizConfigPanel.tsx]: Unnecessary typecast
```suggestion
                });
```

File: packages/front-end/services/dataVizConfigUtilities.ts (hunk 3 above)
Comment (mknowlton89) [reply packages/front-end/services/dataVizConfigUtilities.ts:55]: That's correct.

File: packages/front-end/components/DataViz/SqlExplorerDataVisualization.tsx (hunk 5 above)
Comment (mknowlton89) [reply packages/front-end/components/DataViz/SqlExplorerDataVisualization.tsx:290]: Yup - that is correct. I've got this listed as a fast-follow in Notion for the PA Dashboards project.
